### PR TITLE
S1 t4/update frontend for signup user

### PR DESCRIPTION
**Pull request by sehersarin**
**Scrum Board Ticket:** S1T4: Update Frontend for Signup User 
**Story:** Story 1: Sign up using email 

**Description:** Updated the logic and functionality for the signup form. The signup form enables users to enter their first and last name, select user role, worker type and specialization, enter ID, email, phone number and password to create an account in the system. Frontend unit tests were also written to validate this ticket.

**Testing Criteria (how I tested the code):** 
1. Tested UI of Signup form by running locally and verifying the front-end has matched design specifications. 
2. Tested form input criteria (by creating unit tests) for a variety of instances including:
    1. First name
    2. Last name
    3. Email
    4. Phone number
    5. Password
3. Ran the automated tests in the `react-ui` folder and ensured that there was a high file coverage for the SignupForm component and all tests passed.

**How Reviewers can Test Code:**
1. Run the code locally and ensure no complications when starting up.
2. Navigate inside the `react-ui` folder and run `npm start`.
3. View the screens and UI to ensure it looks like design specifications (view screenshot below.)
Screen Shot 2020-11-04 at 11 19 36 AM

4. Inside the `react-ui folder`, run `npm test`. All tests should pass and the SignupForm component should match the file coverage - as seen in the screenshot below (the rest of the frontend tests are not set up yet and are in the scope for another ticket).
Screen Shot 2020-11-04 at 11 06 15 AM

### CODE REVIEW

File: react-ui/src/components/SignUpForm.js [hunk 1]
```diff
@@ -68,22 +73,47 @@ render() {
...
-                <label for="role" className ="Title-Style2"> Select Role: </label>
-                <select name="role" className ="InputFields2" id="type">
+                <label for="role" className ="Title-Style2"> Select User Role: </label>
+                <select name="role" className ="InputFields3" id="type">
                   <option value="student">Student</option>
+                  <option value="worker1">Worker</option>
+                </select>
+                <br></br>
+                <br></br>
+                <label for="worker_type" className ="Title-Style2"> Select Worker Type (or N/A otherwise): </label>
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:83]: Created a dropdown for "Worker Type" as per design specification document for service workers to select option from drop down menu or select "other" and input their designation in the appropriate input box below. Students are expected to select the N/A option from this dropdown.

File: react-ui/src/components/SignUpForm.js [hunk 2]
```diff
@@ -68,22 +73,47 @@ render() {
...
                       type="text" 
-                      name="optional_role"
-                      placeholder= "Enter Role, If 'Other' Selected" 
+                      name="optional_type"
+                      placeholder= "Enter Worker Type, If 'Other'" 
+                      onChange={this.handleFormChange} />
+                    </label>
+                  <br></br>
+                  <br></br>
+                <label for="specialization" className ="Title-Style2"> Select Worker Specialization (or N/A): </label>
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:102]: Created a dropdown for "Worker Specialization" as per design specification document for service workers to select option from drop down menu or select "other" and input their specialization in the appropriate input box below. Students are expected to select the N/A option from this dropdown.

File: react-ui/src/components/SignUpForm.js [hunk 3]
```diff
@@ -68,22 +73,47 @@ render() {
                 </label>
                 <br></br>
                 <br></br>
-                <label for="role" className ="Title-Style2"> Select Role: </label>
-                <select name="role" className ="InputFields2" id="type">
+                <label for="role" className ="Title-Style2"> Select User Role: </label>
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:76]: User role is a dropdown form option - users are expected to select either Student or Worker when signing up.

File: react-ui/src/components/SignupForm.test.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      const newValue = "World";
+
+    // Act
+    const getByName = queryByAttribute.bind(null, 'name');
+    const lastNameInput = getByName(dom.container, 'last_name');
+    fireEvent.change(lastNameInput, { target: { value: newValue } })
+
+    //Assert
+    expect(lastNameInput.value).toBe(newValue);
+    // console.log('before', lastNameInput.value);
```
Comment (sehersarin) [inline react-ui/src/components/SignupForm.test.js:56]: Left in console.log "debugging" statements for now to assist with future development of frontend unit tests which are still to be setup for other components.

File: react-ui/src/components/SignUpForm.js [hunk 5]
```diff
@@ -68,22 +73,47 @@ render() {
...
+                <select name="role" className ="InputFields3" id="type">
                   <option value="student">Student</option>
+                  <option value="worker1">Worker</option>
+                </select>
+                <br></br>
+                <br></br>
+                <label for="worker_type" className ="Title-Style2"> Select Worker Type (or N/A otherwise): </label>
+                <br></br>
+                <select name="worker_type" className ="InputFields2" id="type">
+                  <option value="student">Social Worker</option>
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:86]: Verified these dropdown values with the invision designs - please let me know if any values are missing/need to be added!

File: react-ui/src/components/SignUpForm.js [hunk 6]
```diff
@@ -68,22 +73,47 @@ render() {
...
+                      name="optional_type"
+                      placeholder= "Enter Worker Type, If 'Other'" 
+                      onChange={this.handleFormChange} />
+                    </label>
+                  <br></br>
+                  <br></br>
+                <label for="specialization" className ="Title-Style2"> Select Worker Specialization (or N/A): </label>
+                <br></br>
+                <select name="specialization" className ="InputFields2" id="type">
+                  <option value="student">Masters in Social Work</option>
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:105]: Verified these dropdown values with the invision designs - please let me know if any values are missing/need to be added!

**Review by prdbondi — APPROVED** [review]
Ran testing steps 1-3. Looks great Seher!

**Comment by sehersarin** [comment]
Thank you for the review Peter!

**Pull request merged**